tests/unit_tests/dft/include/compute_out_of_place.hpp

```diff
@@ -23,73 +23,85 @@
 
-    std::vector<FwdInputType> fwd_data(input);
-
+    auto tmp = std::vector<FwdOutputType>(
+        cast_unsigned(backward_distance * batches + get_default(strides_bwd, 0, 0L)), 0);
     {
         sycl::buffer<FwdInputType, 1> fwd_buf{ fwd_data };
-        sycl::buffer<FwdOutputType, 1> bwd_buf{ sycl::range<1>(
-            cast_unsigned(backward_distance * batches)) };
+        sycl::buffer<FwdOutputType, 1> bwd_buf{ tmp };
 
         oneapi::mkl::dft::compute_forward<descriptor_t, FwdInputType, FwdOutputType>(
             descriptor, fwd_buf, bwd_buf);
 
         {
             auto acc_bwd = bwd_buf.template get_host_access();
             auto bwd_ptr = acc_bwd.get_pointer();
-            auto ref_iter = out_host_ref.begin();
-            const auto ref_row_stride = sizes.back();
-            const auto backward_row_stride = cast_unsigned(get_backward_row_size<domain>(sizes));
-            const auto backward_row_elements = cast_unsigned(get_backward_row_size<domain>(sizes));
-
-            while (ref_iter < out_host_ref.end()) {
-                EXPECT_TRUE(check_equal_vector(bwd_ptr, ref_iter, backward_row_elements,
-                                               abs_error_margin, rel_error_margin, std::cout));
-                bwd_ptr += backward_row_stride;
-                ref_iter += ref_row_stride;
+            for (std::int64_t i = 0; i < batches; i++) {
+                EXPECT_TRUE(check_equal_strided<domain == oneapi::mkl::dft::domain::REAL>(
+                    bwd_ptr + backward_distance * i, out_host_ref.data() + ref_distance * i, sizes,
+                    strides_bwd, abs_error_margin, rel_error_margin, std::cout));
             }
         }
 
-        if constexpr (domain == oneapi::mkl::dft::domain::REAL) {
+        if (strides_bwd.size()) {
+            descriptor.set_value(oneapi::mkl::dft::config_param::INPUT_STRIDES, strides_bwd.data());
+        }
+        else if constexpr (domain == oneapi::mkl::dft::domain::REAL) {
             const auto complex_strides = get_conjugate_even_complex_strides(sizes);
-            auto real_strides = get_default_strides(sizes);
             descriptor.set_value(oneapi::mkl::dft::config_param::INPUT_STRIDES,
                                  complex_strides.data());
+        }
+        else {
+            auto real_strides = get_default_strides(sizes);
+            descriptor.set_value(oneapi::mkl::dft::config_param::INPUT_STRIDES,
+                                 real_strides.data());
+        }
+        if (strides_fwd.size()) {
+            descriptor.set_value(oneapi::mkl::dft::config_param::OUTPUT_STRIDES,
+                                 strides_fwd.data());
+        }
+        else {
+            auto real_strides = get_default_strides(sizes);
             descriptor.set_value(oneapi::mkl::dft::config_param::OUTPUT_STRIDES,
                                  real_strides.data());
-            commit_descriptor(descriptor, sycl_queue);
         }
```

Comment:
`real_` in `real_strides` used here above (and in other places herein) seems to be a fairly confusing misnomer.

Reply:
fixed